src/Sender.cc

```diff
@@ -976,21 +983,49 @@ Sender::checkPingTimeouts()
 }
 
 /**
- * Send out packets for any messages with unscheduled/granted bytes.
+ * Attempt to wake up the pacer thread that is responsible for calling trySend()
+ * repeatedly, if it's currently blocked waiting for packets to become ready to
+ * be sent.
+ *
+ * This method is called when new GRANTs arrive, when new outgoing messages
+ * appear, and when retransmission is requested.
+ *
+ * @param lockHeld
+ *      Reminder to hold the Sender::queueMutex during this call.
  */
 void
-Sender::trySend()
+Sender::signalPacerThread(const SpinLock::Lock& lockHeld)
+{
+    (void)lockHeld;
+    sendReady = true;
+    callbacks->notifySendReady();
```

Comment:
Generally, shouldn't hold the lock while calling a callback; can result in deadlock.

Reply:
This method is now inlined so that `queueMutex` can be released before the calling the callback.